dspace-api/src/main/java/org/dspace/authenticate/SamlAuthentication.java

```diff
@@ -0,0 +1,750 @@
+
+        // Commit the new eperson
+
+        AuthenticateServiceFactory.getInstance().getAuthenticationService().initEPerson(context, request, ePerson);
+
+        ePersonService.update(context, ePerson);
+        context.dispatchEvents();
+
+        // Turn authorizations back on
+
+        context.restoreAuthSystemState();
+
+        if (log.isInfoEnabled()) {
+            String message = "Auto registered new eperson using SAML attributes:\n";
+
+            message += "  netid: " + ePerson.getNetid() + "\n";
+            message += "  email: " + ePerson.getEmail() + "\n";
+            message += "  firstName: " + ePerson.getFirstName() + "\n";
+            message += "  lastName: " + ePerson.getLastName();
+
+            log.info(message);
+        }
+
+        return ePerson;
+    }
+
+    /**
+     * After we successfully authenticated a user, this method will update the user's attributes. The
+     * user's email, name, or other attribute may have been changed since the last time they
+     * logged into DSpace. This method will update the database with their most recent information.
+     *
+     * This method handles the basic DSpace metadata (email, first name, last name) along with
+     * additional metadata set using the setMetadata() methods on the EPerson object. The
+     * additional metadata mappings are defined in configuration.
+     *
+     * @param context The current DSpace database context
+     * @param request The current HTTP Request
+     * @param eperson The eperson object to update.
+     * @throws SQLException       if database error
+     * @throws AuthorizeException if authorization error
+     */
+    protected void updateEPerson(Context context, HttpServletRequest request, EPerson eperson)
+        throws SQLException, AuthorizeException {
+
+        String nameId = findSingleAttribute(request, getNameIdAttributeName());
+
+        String emailAttributeName = getEmailAttributeName();
+        String firstNameAttributeName = getFirstNameAttributeName();
+        String lastNameAttributeName = getLastNameAttributeName();
+
+        String email = findSingleAttribute(request, emailAttributeName);
+        String firstName = findSingleAttribute(request, firstNameAttributeName);
+        String lastName = findSingleAttribute(request, lastNameAttributeName);
+
+        // Truncate values of parameters that are too big.
+
+        if (firstName != null && firstName.length() > NAME_MAX_SIZE) {
+            log.warn(
+                "Truncating eperson's first name because it is longer than " + NAME_MAX_SIZE + ": " + firstName);
+
+            firstName = firstName.substring(0, NAME_MAX_SIZE);
+        }
+
+        if (lastName != null && lastName.length() > NAME_MAX_SIZE) {
+            log.warn("Truncating eperson's last name because it is longer than " + NAME_MAX_SIZE + ": " + lastName);
+
+            lastName = lastName.substring(0, NAME_MAX_SIZE);
+        }
+
+        context.turnOffAuthorisationSystem();
```

Comment:
As per previous comment, this should be in try / catch / finally so we always restore context auth state.

Reply:
Fixed in 8f4aacd.